cmd/parser/parse.go

```diff
@@ -0,0 +1,356 @@
+				signatureKey += "*"
+				negativeSignatureKey += "attr"
+				if result.Rule.PlatformRegion == "*" {
+					// containsWildcards = true
+				}
+			} else {
+				key += result.Rule.PlatformRegion
+			
+				signatureKey += "attr"
+				negativeSignatureKey += "*"
+			}
+			
+			key += "HR:"
+			signatureKey += "HR:"
+			negativeSignatureKey += "HR:"
+			if result.Rule.HyperscalerRegion == "" || result.Rule.HyperscalerRegion == "*" {
+				key += "*"
+			
+				signatureKey += "*"
+				negativeSignatureKey += "attr"
+
+			
+				if result.Rule.HyperscalerRegion == "*" {
+					// containsWildcards = true
+				}
+
+			} else {
+				key += result.Rule.HyperscalerRegion
+			
+				signatureKey += "attr"
+				negativeSignatureKey += "*"
+			}
+
+			negativeSignatureItem, negativeSignatureExists := signatureSet[negativeSignatureKey]
+
+			// if negativeSignatureExists && containsWildcards && cmd.signature {
+			if negativeSignatureExists && cmd.signature {
+				
+				resolvingSignaturePossibleRule := result.Rule.Combine(*negativeSignatureItem.Rule)
+
+			resolvingKey := resolvingSignaturePossibleRule.StringNoLabels()
+			_, resolvingRuleExists := resolvedRules[resolvingKey]
+
+				if !resolvingRuleExists {
+					err := fmt.Errorf("Duplicated negative signature with previously defined rule: '%s', consider introducing a resolving rule '%s'", negativeSignatureItem.Rule.StringNoLabels(), resolvingKey)
+
+					errorResults = append(errorResults, rules.ParsingResult{OriginalRule: result.OriginalRule, Err: err})
+
+					uniqueResults = append(uniqueResults, rules.ParsingResult{OriginalRule: result.OriginalRule, Err: err})
+					continue
+				}
+			}
+
+			alreadyExists := false
+			var item rules.ParsingResult
+			item, alreadyExists = uniqnuessSet[key]
+	
+			if !alreadyExists {
+
+				uniqnuessSet[key] = result
+				signatureSet[signatureKey] = result
+				uniqueResults = append(uniqueResults, result)
+
+			} else {
+				err := fmt.Errorf("Duplicated rule with previously defined rule: '%s'", item.Rule.StringNoLabels())
+
+				errorResults = append(errorResults, rules.ParsingResult{OriginalRule: result.OriginalRule, Err: err})
+
+				uniqueResults = append(uniqueResults, rules.ParsingResult{OriginalRule: result.OriginalRule, Err: err})
+			}
+		}
+
+		allResults = uniqueResults
+	}
+
+	if cmd.sort {
+		allResults = rules.SortRuleEntries(allResults)
+		okResults = rules.SortRuleEntries(okResults)
+		errorResults = rules.SortRuleEntries(errorResults)
+	}
+
+	var testDataForMatching *rules.MatchableAttributes
+	if (cmd.match != "") {
+		testDataForMatching = getTestData(cmd.match)
+	}
+	
+	for _, result := range allResults {
+
+		cmd.cobraCmd.Printf("-> ")
+		if result.Err != nil {
+
+			cmd.cobraCmd.Printf("%s Error %s", colorError, colorNeutral)
+
+		} else {
+
+			if (cmd.match != "" && testDataForMatching != nil) {
+				matched := result.Rule.Matched(testDataForMatching)
+
+				if matched {
+					cmd.cobraCmd.Printf("%s Matched %s ", colorMatched, colorNeutral)
```

Comment:
Could be extracted as function printing messages in given color and reverting neutral one.